tests/robot-tests/tests/public_api/public_api_major_auto_changes.robot

```diff
@@ -0,0 +1,253 @@
+    user checks summary list contains    Indicators    more indicators    parent=id:live-version-summary
+    user checks summary list contains    Filters    Academy type    parent=id:live-version-summary
+    user checks summary list contains    Filters    School type    parent=id:live-version-summary
+    user checks summary list contains    Actions    View live data set (opens in new tab)
+    ...    parent=id:live-version-summary
+
+Validate the summary contents inside the 'draft version details' table
+    user waits until h3 is visible    Draft version details
+    user checks summary list contains    Version    v2.0    parent=id:draft-version-summary    wait=%{WAIT_LONG}
+    user checks summary list contains    Status    Action required    parent=id:draft-version-summary
+    ...    wait=%{WAIT_LONG}
+    user checks summary list contains    Release    ${RELEASE_2_NAME}    parent=id:draft-version-summary
+    user checks summary list contains    Data set file    ${SUBJECT_2_NAME}    parent=id:draft-version-summary
+    user checks summary list contains    Geographic levels    Local authority, National, Regional, School
+    ...    parent=id:draft-version-summary
+    user checks summary list contains    Time periods    2020/21 to 2022/23    parent=id:draft-version-summary
+    user checks summary list contains    Indicators    Enrolments    parent=id:draft-version-summary
+    user checks summary list contains    Indicators    more indicators    parent=id:draft-version-summary
+    user checks summary list contains    Filters    Academy type    parent=id:draft-version-summary
+    user checks summary list does not contain    Filters    School type    parent=id:draft-version-summary
+    user checks summary list contains    Actions    Remove draft version    parent=id:draft-version-summary
+
+Validate that location and filter mappings are complete
+    user waits until h3 is visible    Draft version tasks
+    user waits until parent contains element    testid:map-locations-task    link:Map locations
+    user waits until parent contains element    id:map-locations-task-status    text:Complete
+    user waits until parent contains element    testid:map-filters-task    link:Map filters
+    user waits until parent contains element    id:map-filters-task-status    text:Complete
+
+Confirm finalization of this API data set version
+    user clicks button    Finalise this data set version
+    user waits for caches to expire
+    user waits until h2 is visible    Mappings finalised
+    user waits until page contains    Draft API data set version is ready to be published
+
+User navigates to 'changelog and guidance notes' page and update relevant details in it
+    user clicks link by index    View changelog and guidance notes    1
+    user waits until page contains    API data set changelog
+
+    user enters text into element    css:textarea[id="guidanceNotesForm-notes"]
```

Comment:
Use `id` selector (i.e `id:guidanceNotesForm-notes`) or `name` (i.e. `name:notes`) as mentioned in other comment.

Reply:
Good spot ta.  This was inherited from the originally-written "resolve mappings" suite and I only focused on improving certain bits that stood out as needing tidying.  Scooped these up now too thanks.